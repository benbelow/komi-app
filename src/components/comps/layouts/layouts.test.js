import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'
import { PageWrapper, ContentWrapper, EmptyList } from './index'

describe('layouts animation test suite', () => {
  test('"PageLayout" should match snapshot', () => {
    const comp = renderer.create(
      <PageWrapper>
        <div>
          <p>hello</p>
          <p>world</p>
        </div>
      </PageWrapper>
    )
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('"ListWrapper" should match snapshot', () => {
    const comp = renderer.create(
      <ContentWrapper>
        <div>
          <p>hello</p>
          <p>world</p>
        </div>
      </ContentWrapper>
    )
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('"EmptyList" should match snapshot', () => {
    const comp = renderer.create(
      <StaticRouter context={{ router: { isActive: true } }}>
        <EmptyList message='hello world' to='/tests'>
          <div>
            <p>hello</p>
            <p>world</p>
          </div>
        </EmptyList>
      </StaticRouter>
    )
    const tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
