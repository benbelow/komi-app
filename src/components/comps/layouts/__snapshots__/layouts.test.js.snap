// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts animation test suite "EmptyList" should match snapshot 1`] = `
.c0 {
  color: rgba(0,0,0,.54);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c1 {
  text-align: center;
  padding-top: 80px;
}

<div
  className="c0"
>
  <a
    href="/tests"
    onClick={[Function]}
  >
    <p
      className="c1"
    >
      hello world
    </p>
  </a>
  <div>
    <p>
      hello
    </p>
    <p>
      world
    </p>
  </div>
</div>
`;

exports[`layouts animation test suite "ListWrapper" should match snapshot 1`] = `
.c0 {
  background-color: #fafafa;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100vw;
  max-width: 460px;
  min-height: calc(100vh - 50px);
  padding-bottom: 70px;
}

<div
  className="c0"
>
  <div>
    <p>
      hello
    </p>
    <p>
      world
    </p>
  </div>
</div>
`;

exports[`layouts animation test suite "PageLayout" should match snapshot 1`] = `
.c0 {
  background-color: #f3e5f5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: top;
  -webkit-justify-content: top;
  -ms-flex-pack: top;
  justify-content: top;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 50px;
  border: none;
  width: 100vw;
  min-height: 100vh;
}

<div
  className="c0"
>
  <div>
    <p>
      hello
    </p>
    <p>
      world
    </p>
  </div>
</div>
`;
